crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1320,6 +1320,23 @@ impl OlmMachine {
         })
     }
 
+    async fn get_megolm_encryption_info(
+        &self,
+        room_id: &RoomId,
+        event: &EncryptedEvent,
+        content: &SupportedEventEncryptionSchemes<'_>,
+    ) -> MegolmResult<EncryptionInfo> {
+        if let Some(session) =
+            self.store().get_inbound_group_session(room_id, content.session_id()).await?
+        {
+            self.get_encryption_info(&session, &event.sender).await
+        } else {
+            // don't bother to check for withheld codes here: the client should only call
+            // this on events that were successfully decrypted anyway.
+            Err(MegolmError::MissingRoomKey(None))
```

Comment:
Couldn't this method be useful in the case a withheld code has been received after the initial decryption attempt failed or a better withheld code has been received? 

In that case you also wouldn't want to do the decryption dance to update the message associated to the UTD.

Reply:
done in https://github.com/matrix-org/matrix-rust-sdk/pull/2510/commits/3793c527d427e2f06682b18207b9d6314e57c36d?diff=unified&w=1. Suggest viewing with whitespace hidden!